manual/embedded-examples/src/docs/dev/index.asciidoc

```diff
@@ -4,6 +4,17 @@
 It's easy to use Neo4j embedded in Java applications.
 In this chapter you will find everything needed -- from setting up the environment to doing something useful with your data.
 
+[WARNING]
+====
+When running your own code alongside Neo4j in the same JVM, there are a few things you should keep in mind:
+
+* Don't create or retain more objects than you strictly need to.
+  Large caches are particularly good at promoting many objects to the old generation, thus increasing the need for expensive full garbage collections.
```

Comment:
Hmm, saying "Large caches are particularly good..." sounds like an advice rather than a warning.
Can we say "Large caches cause the promotion of ..." instead?